src/main/java/com/formulasearchengine/mathmlsim/similarity/MathPlag.java

```diff
@@ -0,0 +1,107 @@
+        // Find matches between them. Single leafs are (e.g. a single variable) will not be considered
+        return new SubTreeComparison(type).getSimilarities(refDoc, compDoc, true);
+    }
+
+    /**
+     * Compare two CMMLInfo document against each other. The type defines how a Match will be displayed.
+     *
+     * @param refMathML  Reference MathML string (must contain pMML and cMML)
+     * @param compMathML Comparison MathML string (must contain pMML and cMML)
+     * @return list of matches / similarities, list can be empty.
+     */
+    public static List<Match> compareSimilarMathML(String refMathML, String compMathML) throws IOException, ParserConfigurationException {
+        MathNode refMathNode = MathNodeGenerator.generateMathNode(new CMMLInfo(refMathML)).toAbstract();
+        MathNode compMathNode = MathNodeGenerator.generateMathNode(new CMMLInfo(compMathML)).toAbstract();
+        return new SubTreeComparison("similar").getSimilarities(refMathNode, compMathNode, true);
+    }
+
+    /**
+     * Old score - this is basically a printMathNode out for comparison and be deleted later on
+     *
+     * @param refMathML  Reference MathML string (must contain pMML and cMML)
+     * @param compMathML Comparison MathML string (must contain pMML and cMML)
+     * @return map of all found factors
+     * @throws ParserConfigurationException malformed mathml or even xml in most cases
+     * @throws XPathExpressionException could hint towards a bug
+     * @throws IOException transformation exception between document and string
+     */
+    public static Map<String, Object> compareOriginalFactors(String refMathML, String compMathML) throws ParserConfigurationException, XPathExpressionException, IOException {
+        try {
+            CMMLInfo refDoc = new CMMLInfo(refMathML);
+            CMMLInfo compDoc = new CMMLInfo(compMathML);
+            // compute factors
+            final Integer depth = compDoc.getDepth(refDoc.getXQuery());
+            final Double coverage = compDoc.getCoverage(refDoc.getElements());
+            Boolean formula = compDoc.isEquation(true);
+            Boolean structMatch = compDoc.toStrictCmml().abstract2CDs()
+                    .isMatch(refDoc.toStrictCmml().abstract2CDs().getXQuery());
+            Boolean dataMatch = new CMMLInfo(compMathML).toDataCmml().abstract2DTs()
+                    .isMatch(new CMMLInfo(refMathML).abstract2DTs().getXQuery());
+
+            HashMap<String, Object> result = new HashMap<>();
+            result.put("depth", depth);
+            result.put("coverage", coverage);
+            result.put("structure match", structMatch);
+            result.put("data match", dataMatch);
+            result.put("formula", formula);
+            return result;
+        } catch (Exception e) {
+            // log and throw in this case
+            logger.error("mathml comparison failed", e);
```

Comment:
For debug and reply it would be super nice to have refMathML and compMathML available from the error message.